src/main/java/net/dv8tion/jda/internal/managers/channel/ChannelManagerImpl.java

```diff
@@ -605,7 +605,7 @@ public M setAvailableTags(List<? extends BaseForumTag> tags)
     public M setAppliedTags(Collection<? extends ForumTagSnowflake> tags)
     {
         if (type != ChannelType.GUILD_PUBLIC_THREAD)
-            throw new IllegalStateException("Can only set applied tags on public thread channels.");
+            throw new IllegalStateException("Can only set applied tags on forum post thread channels.");
```

Comment:
Can you call this method on normal thread creation? this if wouldn't catch that, right?
We should probably check to make sure the parent is actually a ForumChannel

Reply:
We do check it further down